WooCommerce/Classes/ViewModels/InAppPurchases/UpgradesViewModel.swift

```diff
@@ -92,11 +94,11 @@ final class UpgradesViewModel: ObservableObject {
     @MainActor
     func purchasePlan(with planID: String) async {
         analytics.track(event: .InAppPurchases.planUpgradePurchaseButtonTapped(planID))
-        guard let wooWPComPlan = planCanBePurchasedFromCurrentState() else {
+        guard let wooWPComPlan = planCanBePurchasedFromCurrentState(with: planID) else {
             return
         }
 
-        upgradeViewState = .purchasing(wooWPComPlan)
+        upgradeViewState = .purchasing(wooWPComPlan, plans)
```

Comment:
Just leaving a note here for reference: I haven't tested this fully, and not sure if we would need it, but if we happen to not want both associated values to the .purchasing state we could also retrieve `plans` from the `planCanBePurchasedFromCurrentState` method above if we return a tuple, sort of:

```
func planCanBePurchasedFromCurrentState(with planID: String) -> (WooWPComPlan?, [WooWPComPlan])? {
    switch upgradeViewState {
    case .loaded(let plans):
        let selectedPlan = plans.first(where: { $0.wpComPlan.id == planID })
        return (selectedPlan, plans)
    case .purchaseUpgradeError(.inAppPurchaseFailed(let plan, _)):
        return (plan, [])
    default:
        return nil
    }
}
```